python_benchmarks/qr_codes/qr.py

```diff
@@ -4,21 +4,75 @@
 Usage example:
 python qr.py -o out.yaml -p qrcodes/detection
 -H, --help - show help
--o, --output - output file (default out.yaml)
+-o, --output - output path
 -p, --path - input dataset path (default qrcodes/detection)
+--per_image_statistic - print the per image statistic
 -a, --accuracy - input accuracy (default 20 pixels)
 -alg, --algorithm - input alg (default opencv)
 --metric - input norm (default l_inf)
 """
 
+import time
+import cv2 as cv
+import numpy as np
+from numpy import linalg as LA
+from collections import OrderedDict
 import argparse
+import os
 import glob
+import pandas as pd
+import datetime
+import matplotlib.pyplot as plt
+import json
 from enum import Enum
+from iteration_utilities import deepflatten
 
-import time
-import numpy as np
-from numpy import linalg as LA
-import cv2 as cv
+# l_1 - https://en.wikipedia.org/wiki/Norm_(mathematics)
+# l_inf - Chebyshev norm https://en.wikipedia.org/wiki/Chebyshev_distance
+TypeNorm = Enum('TypeNorm', 'l1 l2 l3 l_inf intersection_over_union')
+
+
+def get_max_error(accuracy, metric):
+    if metric is TypeNorm.l1 or metric is TypeNorm.l2 or metric is TypeNorm.l_inf:
+        return 2 * accuracy
+    if metric is TypeNorm.intersection_over_union:
+        return 1.
+    raise TypeError("this TypeNorm isn't supported")
+
+
+def get_norm(gold_corner, corner, metric):
+    if metric is TypeNorm.l1:
+        return LA.norm((gold_corner - corner).flatten(), 1)
+    if metric is TypeNorm.l2 or metric is TypeNorm.intersection_over_union:
+        return LA.norm((gold_corner - corner).flatten(), 2)
```

Comment:
I propose to use name for parameter `ord`. Current `1` and `2`, etc is not obvious for reader.

Reply:
added